internal/retry/sizer.go

```diff
@@ -0,0 +1,87 @@
+package retry
+
+// SizerCheckFunc allows the creator of a Sizer to specify arbitrarily
+// complex checks of the error condition to determine if the size
+// needs to be retried.
+type SizerCheckFunc func(error) bool
+
+// Sizer is used to implement 'resize loops' that hides the complexity
+// of the sizing and error checking away from most of the application.
+//
+// At the end of each iteration call Update or UpdateWants to inform
+// the sizer of the result of the action(s) taken during the iteration.
+// If the error is nil the update functions will return nil and halt
+// the iteration. If the error is non nil but returns false from
+// the check function the same error is returned and iteration will halt.
+// If the check function returns true iteration continues but nil
+// will be returned from the update function, effectively clearing
+// the error state, *unless* the current size now exceeds maxSize.
+type Sizer struct {
```

Comment:
Consider calling it Grower or Increaser?

Reply:
Unfortunately, neither of those appeal to me and they're both longer.
For some reason this work causes me to have a mental image of a tailor sizing someone up for clothes. :-)
I was going to do either base the name on the word Size or Fit. I went with size -> Sizer.